tests/dispatch_io_pipe.c

```diff
@@ -404,7 +404,14 @@ test_dispatch_write(int kind, int delay)
 	dispatch_group_t g = dispatch_group_create();
 	dispatch_group_enter(g);
 
-	const size_t bufsize = test_get_pipe_buffer_size(kind);
+	// We must subtract 1 from the bufsize because the write operation
+	// uses 1 byte of the buffer as a sentinel value to ensure that a
+	// reader is making progress. Because these tests operate serially,
```

Comment:
It might be easier to follow if we describe the libdispatch behavior rather than the rationale of its implementation:

> The libdispatch implementation writes at most `bufsize-1` before requiring the reader to start draining the pipe.